src/components/ebay-pagination/index.js

```diff
@@ -91,46 +92,42 @@ function refresh() {
     for (let i = 0; i < this.state.items.length; i++) {
         if (this.state.items[i].current) {
             current = i;
-        } else {
-            // remove all hidden attribues to get accurate widths
-            this.pageEls[i].removeAttribute('hidden');
         }
+        this.pageEls[i].removeAttribute('hidden');
     }
+
     const totalPages = this.pageEls.length;
-    const pageNumWidth = this.pageEls[current].offsetWidth + constants.margin;
-    const containerWidth = this.containerEl.offsetWidth - pageNumWidth * 2;
-    const numPagesAllowed = Math.floor((containerWidth) / (pageNumWidth));
-    const adjustedNumPages = Math.min(constants.maxPagesAllowed - 1,
-        Math.max(numPagesAllowed, constants.minPagesRequired));
+    const pageNumWidth = this.pageEls[0].children[0].offsetWidth + constants.margin;
+    const numPagesAllowed = (((this.pageContainerEl.offsetWidth) / pageNumWidth));
+    const adjustedNumPages = Math.floor(Math.min(constants.maxPagesAllowed,
+        Math.max(numPagesAllowed, constants.minPagesRequired)));
 
     let start = 0;
     let end = adjustedNumPages;
-    const rangeLeft = Math.floor(adjustedNumPages * 0.5);
+    let rangeLeft = Math.floor(adjustedNumPages * 0.5);
     const rangeRight = Math.floor(adjustedNumPages * 0.5);
 
+    if (rangeLeft + rangeRight + 1 > adjustedNumPages) {
+        rangeLeft -= 1;
```

Comment:
Make sure that this new logic path is covered with a new test case.

Reply:
I will write a new test for this case after lunch.  just need to add a new width

Reply:
there is no a new test at width 440px when the fifth element is selected.  It only works when the current element is not near the beginning or end.